src/stactools/sentinel2/stac.py

```diff
@@ -464,6 +516,17 @@ def metadata_from_granule_metadata(
         os.path.join(granule_metadata_href, "tileInfo.json"), read_href_modifier
     )
 
+    product_metadata = None
+    if os.path.exists(f := os.path.join(granule_metadata_href, "product_metadata.xml")):
```

Comment:
this handles the cases of 
1. It's running as part of test or you've renamed the product metdata file (metadata.xml) to product_metadata.xml
2. It's running aginst the SH api, so we retrieve it.
Otherwise, there's not product metadata file and we use the processing baseline to guess at the values